services/QuillLMS/client/app/bundles/PremiumHub/containers/diagnosticGrowthReports/helpers.tsx

```diff
@@ -93,8 +96,17 @@ function overallSkillGrowthValue({diagnosticId, overallSkillGrowth, handleGrowth
   return 'No Growth';
 }
 
-function createAggregateRowData({ aggregateRowsDataForDiagnostic, diagnosticId, handleGrowthChipClick, handlePreDiagnosticChipClick }) {
-  return Object.keys(aggregateRowsDataForDiagnostic).map(key => {
+export function createAggregateRowData({ aggregateRowsDataForDiagnostic, diagnosticId, handleGrowthChipClick, handlePreDiagnosticChipClick, groupByValue }) {
+  let keys
+  if (groupByValue === 'teacher' || groupByValue === 'classroom') {
```

Comment:
I think that the back-end should also have the intended order when sorted by "grade" as well, so you can probably get away with skipping this conditional and just always calling the `sort` function?  I suppose that if grade is already in the correct order that would actually involve more work on the front-end by sorting when it's not fully necessary, but the code would be easier to read without the extra conditional, so I think it's just a judgment call.